lib/errdiff/errdiff.go

```diff
@@ -0,0 +1,34 @@
+package errdiff
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Substring checks that an error `got` matches the expectation `want`.
+// Empty `want` string means no error is expected; if not empty, `got` must be
+// non-nil and contain `want` as a substring to meet the expectation.
+// Returns an empty string if `got` meets the `want` expectation, or a string
+// containing an explanation of the discrepancy otherwise.
+//
+// Example use:
+//
+//   if diff := errdiff.Substring(gotErr, "some substring"); diff != "" {
+//     t.Error(diff)
+//   }
+func Substring(got error, want string) string {
```

Comment:
So honestly I would nix this whole PR, this already exists within the errors package by default. 

https://cs.opensource.google/go/go/+/refs/tags/go1.17.1:src/errors/wrap.go;l=39

Also generally, don't be afraid of passing in a *testing.T to a helper function for testing. It's a good way of soft locking a function to just test.

Reply:
Just some example code on how to use it with fmt.Errorf
```
	somethingErr := errors.New("library: something error")
	myErr := fmt.Errorf("err: %w %s", somethingErr, "my path")
	fmt.Println(errors.Is(myErr, somethingErr)) //true
```